src/systemtest/java/reposense/ConfigSystemTest.java

```diff
@@ -86,7 +86,8 @@ private void generateReport(String inputDates)
         RepoConfiguration.setDatesToRepoConfigs(
                 repoConfigs, cliArguments.getSinceDate(), cliArguments.getUntilDate());
 
-        ReportGenerator.generateReposReport(repoConfigs, FT_TEMP_DIR, TEST_REPORT_GENERATED_TIME);
+        ReportGenerator.generateReposReport(repoConfigs, FT_TEMP_DIR, TEST_REPORT_GENERATED_TIME,
+                cliArguments.getSinceDate().orElse(null), cliArguments.getUntilDate().orElse(null));
```

Comment:
if not provided, should retrieve the since date from the configs, such as the earliest commit of all repos.

Reply:
It is handled automatically by other parts of the code I believe. The JSON file all this time does not have sinceDate if it is not specified

Reply:
Yes, its handled by the frontend. We should move it to the back as browser has limited ram and processing power.